spec/requests/api/lesson_completions_spec.rb

```diff
@@ -0,0 +1,116 @@
+      end
+    end
+
+    context 'when authenticated' do
+      let(:authentication_header) do
+        {
+          'HTTP_AUTHORIZATION' => ActionController::HttpAuthentication::Basic.encode_credentials(
+            'development',
+            'qwerty123'
+          )
+        }
+      end
+
+      it 'returns all lesson completions after the start date when only the start date is provided' do
+        lesson = create(:lesson)
+        create(:lesson_completion, created_at: 2.days.ago, lesson:)
+        create(:lesson_completion, created_at: 3.days.ago, lesson:)
+        create(:lesson_completion, created_at: 5.days.ago, lesson:)
+
+        get api_lesson_completions_path, params: { start_date: 4.days.ago.to_s }, headers: authentication_header
+
+        expect(JSON.parse(response.body).first).to match(
+          a_hash_including(
+            'sections' => [
+              {
+                'title' => lesson.section.title,
+                'lessons' => [
+                  'title' => lesson.title,
+                  'completions' => 2
+                ]
+              }
+            ]
+          )
+        )
+      end
+
+      it 'returns all lesson completions before the end date when only the end date is provided' do
+        lesson = create(:lesson)
+        create(:lesson_completion, created_at: 2.days.ago, lesson:)
+        create(:lesson_completion, created_at: 3.days.ago, lesson:)
+        create(:lesson_completion, created_at: 5.days.ago, lesson:)
+
+        get api_lesson_completions_path, params: { end_date: 4.days.ago.to_s }, headers: authentication_header
+
+        expect(JSON.parse(response.body).first).to match(
+          a_hash_including(
+            'sections' => [
+              {
+                'title' => lesson.section.title,
+                'lessons' => [
+                  'title' => lesson.title,
+                  'completions' => 1
+                ]
+              }
+            ]
+          )
+        )
+      end
+
+      it 'returns all lesson completions between the start and end dates when both dates are provided' do
+        lesson = create(:lesson)
+        create(:lesson_completion, created_at: 2.days.ago, lesson:)
+        create(:lesson_completion, created_at: 4.days.ago, lesson:)
+        create(:lesson_completion, created_at: 6.days.ago, lesson:)
+        create(:lesson_completion, created_at: 11.days.ago, lesson:)
+
+        get(
+          api_lesson_completions_path,
+          params: { start_date: 10.days.ago.to_s, end_date: 3.days.ago.to_s },
+          headers: authentication_header
+        )
+
+        expect(JSON.parse(response.body).first).to match(
+          a_hash_including(
+            'sections' => [
+              {
+                'title' => lesson.section.title,
+                'lessons' => [
+                  'title' => lesson.title,
+                  'completions' => 2
+                ]
+              }
+            ]
+          )
+        )
+      end
+    end
+
```

Comment:
**question (non-blocking):**  Just checking, but do we no longer need the test for missing start and end dates from the old spec?

Original controller spec:
`context 'when start and end dates are not present' do`
`it 'uses the default dates' do`

Reply:
Thanks @dm-murphy, Great spot 🦅 👁️

Reply:
Thanks! Happy to help!